CleverPush/Source/CleverPushInstance.m

```diff
@@ -385,8 +386,10 @@ - (void)initWithChannelId {
     if (subscriptionId != nil) {
         [self areNotificationsEnabled:^(BOOL notificationsEnabled) {
             if (!notificationsEnabled && !ignoreDisabledNotificationPermission) {
-                [CPLog info:@"notification authorization revoked, unsubscribing"];
-                [self unsubscribe];
+                if (autoRequestNotificationPermission) {
```

Comment:
Not required here for unsubscribe

Reply:
![Screenshot 2023-11-21 at 4 18 09 PM](https://github.com/cleverpush/cleverpush-ios-sdk/assets/110810122/c1c45278-d1f2-4032-bd6c-cc382261022d)

Here is the scenario for that:

1. The user sets the `autoRequestNotificationPermission` to false.
2. We will call the sync subscription API, and we will get the subscription ID.
3. After that, the user will kill the application and launch it again.
4. Then it will go into this block because the user has not given the push notification permission, and we will get false from the `areNotificationsEnabled` method. That's why every second, if `autoRequestNotificationPermission` == false, the user will unsubscribe automatically.

That's why I kept the condition here. (Either every time a new subscription ID has been generated.)